packages/openid4vc/src/shared/utils.ts

```diff
@@ -49,21 +51,62 @@ async function getKeyFromDid(agentContext: AgentContext, didUrl: string) {
   return getKeyFromVerificationMethod(verificationMethod)
 }
 
-export function getVerifyJwtCallback(agentContext: AgentContext): VerifyJwtCallback {
+type VerifyJwtCallbackOptions = {
+  federation?: {
+    trustedEntityIds?: string[]
+  }
+}
+
+export function getVerifyJwtCallback(
+  agentContext: AgentContext,
+  options: VerifyJwtCallbackOptions = {}
+): VerifyJwtCallback {
   return async (jwtVerifier, jwt) => {
     const jwsService = agentContext.dependencyManager.resolve(JwsService)
+
     if (jwtVerifier.method === 'did') {
       const key = await getKeyFromDid(agentContext, jwtVerifier.didUrl)
       const jwk = getJwkFromKey(key)
 
       const res = await jwsService.verifyJws(agentContext, { jws: jwt.raw, jwkResolver: () => jwk })
       return res.isValid
-    } else if (jwtVerifier.method === 'x5c' || jwtVerifier.method === 'jwk') {
+    }
+
+    if (jwtVerifier.method === 'x5c' || jwtVerifier.method === 'jwk') {
       const res = await jwsService.verifyJws(agentContext, { jws: jwt.raw })
       return res.isValid
-    } else {
-      throw new Error(`Unsupported jwt verifier method: '${jwtVerifier.method}'`)
     }
+
+    if (jwtVerifier.method === 'openid-federation') {
+      const { entityId } = jwtVerifier
+      const trustedEntityIds = options.federation?.trustedEntityIds
+      if (!trustedEntityIds)
+        throw new CredoError('No trusted entity ids provided but is required for the openid-federation method.')
+
+      const validTrustChains = await resolveTrustChains({
+        entityId,
+        trustAnchorEntityIds: trustedEntityIds,
+        verifyJwtCallback: async ({ data, signature, jwk }) => {
+          const jws = `${TypedArrayEncoder.toUtf8String(data)}.${TypedArrayEncoder.toBase64URL(signature)}`
+
+          const res = await jwsService.verifyJws(agentContext, {
+            jws,
+            jwkResolver: () => getJwkFromJson(jwk),
+          })
+
+          return res.isValid
+        },
+      })
+
+      // TODO: There is no check yet for the policies
+
+      // TODO: When this function results in a `false` it gives a really misleading error message: 'Error verifying the DID Auth Token signature.'
```

Comment:
we should probably add some logging in this method at least. This is the error from Sphereon lib